src/platform_impl/macos/app_delegate.rs

```diff
@@ -1,10 +1,20 @@
-use super::{activation_hack, app_state::AppState};
-use cocoa::base::id;
+use crate::event::Event;
+
+use super::{activation_hack, app_state::AppState, event::EventWrapper};
+use cocoa::{
+    base::id,
+    foundation::{NSArray, NSString},
+};
 use objc::{
     declare::ClassDecl,
-    runtime::{Class, Object, Sel},
+    runtime::{Class, Object, Sel, BOOL, YES},
 };
-use std::os::raw::c_void;
+use std::{os::raw::c_void, slice, str};
+
+/// This constant represents the one with an identical name in appkit. See:
+/// https://developer.apple.com/documentation/appkit/nsapplicationdelegatereply/nsapplicationdelegatereplysuccess?language=occ
+#[allow(non_upper_case_globals)]
+const NSApplicationDelegateReplySuccess: i32 = 0;
```

Comment:
The [`ffi`](https://github.com/rust-windowing/winit/blob/master/src/platform_impl/macos/ffi.rs) module is our designated dumping ground for constants and such.